$NetBSD: patch-eq,v 1.1 2010/10/27 13:24:09 tsutsui Exp $

Pull fixes for ESP SCSI DMA from upstream master:
http://git.qemu.org/qemu.git/commit/?id=73d7434279e390505164afd02360eebe4b43c7fa

Also disable floppy which is broken.

--- hw/mips_jazz.c.orig
+++ hw/mips_jazz.c
@@ -136,7 +136,7 @@
     NICInfo *nd;
     PITState *pit;
     DriveInfo *fds[MAX_FD];
-    qemu_irq esp_reset;
+    qemu_irq esp_reset, dma_enable;
     qemu_irq *cpu_exit_irq;
     ram_addr_t ram_offset;
     ram_addr_t bios_offset;
@@ -244,7 +244,7 @@
     /* SCSI adapter */
     esp_init(0x80002000, 0,
              rc4030_dma_read, rc4030_dma_write, dmas[0],
-             rc4030[5], &esp_reset);
+             rc4030[5], &esp_reset, &dma_enable);
 
     /* Floppy */
     if (drive_get_max_bus(IF_FLOPPY) >= MAX_FD) {
@@ -254,7 +254,9 @@
     for (n = 0; n < MAX_FD; n++) {
         fds[n] = drive_get(IF_FLOPPY, 0, n);
     }
+#if 0	/* XXX: floppy emulation uses wrong ISA DMA (should be jazz DMA) */
     fdctrl_init_sysbus(rc4030[1], 0, 0x80003000, fds);
+#endif
 
     /* Real time clock */
     rtc_init(1980, NULL);
